ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util.hs

```diff
@@ -63,10 +66,21 @@ module Ouroboros.Consensus.Util (
     -- * Miscellaneous
   , eitherToMaybe
   , fib
+    -- * Fuse code
+  , Fuse
+  , FuseCodeAccessedConcurrently (..)
+  , WithIntactFuse
```

Comment:
This comment is about naming, so take a deep breath :D

I propose these names:

```
-- | A simple semaphore, though instead of blocking a fatal exception is thrown.
newFuse :: String -> m (Fuse m)

-- | An action that cannot be ran without drawing current through a 'Fuse'.
Electric :: (Type -> Type) -> Type -> Type

-- | Too much electrical lode was put on the 'Fuse', see 'drawCurrent'.
FuseBlownException :: Type

-- | See 'Electric'
electric :: m a -> Electric m a

-- | Put full load on the 'Fuse' while the 'Electric' is running.
--
-- Thus any two 'withFuse' calls with the same 'Fuse' will throw one fatal exception.
--
-- NOTE The metaphor is: when I run at most one waffle iron concurrently, my kitchen's fuse doesn't blow. But it blows if I run more than one waffle iron concurrently.
--
-- WARNING If the given action throws its own exception, then it will never stop putting load on the 'Fuse'.
withFuse :: Fuse m -> Electric m a -> m a
```